packages/job-worker/src/workers/parent-base.ts

```diff
@@ -187,8 +185,16 @@ export abstract class WorkerParentBase {
 									continue
 								case ThreadStatus.PendingInit:
 									logger.debug(`Re-initialising worker thread: "${this.#queueName}"`)
-									// Reinit the worker
-									await this.initWorker(mongoUri, this.#mongoDbName)
+									// Reinitialize the worker
+									try {
+										await this.initWorker(mongoUri, this.#mongoDbName)
+									} catch (error) {
+										logger.error(`Worker ${this.#prettyName} could not be initialized:`, {
+											data: error,
+										})
+										this.#threadStatus = ThreadStatus.Closed
+										break
```

Comment:
Shoudn't this be a `continue`? Otherwise we're going to fall through, the thread might try to pick up a job as if the status was `Ready`, even though it's not restarted yet => more problems.

Reply:
@ianshade That is correct. It has been changed to `continue`.